### ISSUE TITLE
Mat.setTo() method is not working. And opencv.core.add() does not have enough parameters

### ISSUE THREAD

**Issue by Bahramudin** [bytedeco/javacv#738]
I am trying to set the value of the Mat to 255 (white), it throws an exception. 

It is the code:
```
Mat mask = Mat.zeros(frame.size(), CV_8UC1).asMat();
Mat maskSubMat = new Mat(mask, faceRect);
maskSubMat .setTo(new Mat(maskSubMat.size(), maskSubMat.type(), Scalar.WHITE));
```
The Exception:

> OpenCV Error: Assertion failed (checkScalar(value, type(), _value.kind(), _InputArray::MAT )) in cv::Mat::setTo, file C:\projects\bytedeco\javacpp-presets\opencv\cppbuild\windows-x86_64\opencv-3.2.0\modules\core\src\copy.cpp, line 538

This API should not receive a Mat object, it will good to pass a Scalar object, something like this:
`mat.setTo(Scalar scalar);`

And also the the opencv_core.add() parameters are not enough, it should be something like this:
```
void add(Mat src, Scalar scalar, Mat result, Mat mask)
// or
void add(Mat src1, Mat src2, Mat result, Mat mask)
```

**It is the original OpenCV API (simple, easy, beautiful and enjoyable)**
```
Mat mask = Mat.zeros(frame.size(), CvType.CV_8UC1);
Mat maskSubMat = mask.submat(faceRect);
maskSubMat.setTo(new Scalar(255));
Core.add(frame, new Scalar(90, 90, 90), frame, mask);
```

**Comment by saudet** [comment]
Try something like this: https://github.com/bytedeco/javacv/issues/684#issuecomment-297571167

We need to add at least a constructor for Mat that takes a Scalar as single argument...

**Comment by saudet** [comment]
The new convenience constructors are now part of the newly released version 1.4. Enjoy!
